### PR TITLE
Verify email toast and extension first privacy modal 

### LINKED ISSUE #374: Verify Email - toast

### LINKED ISSUE #375: Verify email modal

### LINKED ISSUE #377: Privacy Policy - toast

### LINKED ISSUE #378: Add extension - toast

### PR DESCRIPTION
**Pull request by Carla-Moz**
This PR completes the following major tasks:

1. It adds privacy modal to the products page for the extension first onboarding flow
2.  Adds the verify email, privacy policy, and add extension toasts
4. Adds the check your email modal 

@tangcasey, I changed the position of the toast because I don't think they should block the user dropdown icon.

Screen Shot 2022-10-11 at 10 57 40 PM

Screen Shot 2022-10-10 at 7 29 37 PM

Screen Shot 2022-10-11 at 4 40 03 PM

@aaga, you can where I put the product toasts and alerts in `rally/web-platform/website/src/components/pages/studies/ProductAlerts.tsx` and can update Strings in `rally/web-platform/website/src/resources/Strings.tsx`

closes #374 
closes #375
closes #378 
closes #377

### CODE REVIEW

File: web-platform/website/src/pages/login.tsx [hunk 1]
```diff
@@ -34,7 +34,9 @@ const LoginPage: NextPage = () => {
       router.replace("/get-extension");
       return null;
     }
-  } else if (user) {
+  }
+
+  if (user) {
```
Comment (aaga) [inline web-platform/website/src/pages/login.tsx:39]: just want to double check that this logic change is intended

Comment (Carla-Moz) [reply web-platform/website/src/pages/login.tsx:39]: Yes, it was.

**Comment by thomik-corp** [comment]
We already have exclaimation icon in public/img in warning.svg.

File: web-platform/website/src/styles/Colors.ts [hunk 2]
```diff
@@ -1,143 +1,144 @@
...
-  ColorMozWarmRed = "#ff4f5e",
-  ColorMozNeonGreen = "#54ffbd",
-  ColorMozDarkPurple = "#6e008b",
-  ColorMozDarkGreen = "#005e5e",
-  ColorMozDarkBlue = "#00458b",
-  ColorMozDarkGray = "#959595",
-  ColorMozLightGray = "#e7e5e2",
-  ColorHelperGray = "#42435A",
-  ColorTransparent = "transparent",
+	ColorGreen05 = "#e3fff3",
```
Comment (aaga) [inline web-platform/website/src/styles/Colors.ts:2]: @Carla-Moz btw I think this file needs to be re-saved with the "2 spaces" tabbing

Comment (Carla-Moz) [reply web-platform/website/src/styles/Colors.ts:2]: GAH!

**Comment by thomik-corp** [comment]
We also have green checkbox icon in img/check-circle.svg.

File: web-platform/website/src/components/pages/studies/ProductToasts.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+          />
+          <div>{text}</div>
+        </div>
+
+        <div className="right d-flex align-items-center">
+          <Button
+            onClick={async () => {
+              openModal ? await sendEmailVerification() : null;
+              openModal && (openModal as () => void)();
+              openPrivacyModal && (openPrivacyModal as () => void)();
```
Comment (thomik-corp) [inline web-platform/website/src/components/pages/studies/ProductToasts.tsx:120]: Should define types as function and not do typecasting here.

**Comment by jepstein** [comment]
I'm not seeing the email verification modal when signing up with a new email account.

**Comment by Carla-Moz** [comment]
> > I'm not seeing the email verification modal when signing up with a new email account.
> 
> Could you please post a picture of where you're not seeing this?

**Comment by jepstein** [comment]
> 

I get the toast, but the modal doesn't appear. 

https://user-images.githubusercontent.com/444356/195375849-d3d931d9-8057-429f-82f1-d7432d2b714d.mov

**Comment by Carla-Moz** [comment]
> > 
> 
> I get the toast, but the modal doesn't appear.
> Screen.Recording.2022-10-12.at.10.49.34.AM.mov

@jepstein The "check your email" modal is supposed to appear after you hit "Resend Email" on the "verify your email "toast. Could you please try pressing the "Resend" button on the toast?

File: web-platform/website/src/components/pages/privacy-policy/PrivacyPolicyPageContentV2.tsx [hunk 4]
```diff
@@ -22,16 +22,14 @@ import { PrivacyPolicySharing } from "./PrivacyPolicySharing";
 const strings = Strings.components.pages.privacyPolicy;
 const btnStrings = Strings.components.pages.privacyPolicy.buttons;
 
-export function PrivacyPolicyPageContentV2(props: {
-  closeModal: object;
-  isOpen: boolean;
-}) {
+export function PrivacyPolicyPageContentV2(props: { closeModal: object }) {
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply web-platform/website/src/components/pages/privacy-policy/PrivacyPolicyPageContentV2.tsx:25]: Actually, I've decided to handle the closeModal function within the modal instead. It's a lot simpler.

**Comment by jepstein** [comment]
> > > 
> > 
> > 
> > I get the toast, but the modal doesn't appear.
> > Screen.Recording.2022-10-12.at.10.49.34.AM.mov
> 
> @jepstein The "check your email" modal is supposed to appear after you hit "Resend" on the "verify your email" toast. Could you please try pressing the "Resend" button on the toast?

Aha! Didn't realize that's how it worked. Sorry for the misdirection. It's working just as expected.

File: web-platform/website/src/components/pages/studies/ProductToasts.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+import { detectBrowser } from "../../../utils/BrowserDetector";
+import { BrowserType } from "../../../utils/BrowserType";
+
+const emailStrings = Strings.components.pages.studies.alerts.verifyEmail;
+const privacyStrings = Strings.components.pages.studies.alerts.privacyPolicy;
+const extensionStrings = Strings.components.pages.studies.alerts.addExtension;
+
+export function ProductToasts(props: {
+  openModal: object;
+  openPrivacyModal: object;
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply web-platform/website/src/components/pages/studies/ProductToasts.tsx:21]: Gotcha!

**Comment by Carla-Moz** [comment]
> > > > 
> > > 
> > > 
> > > I get the toast, but the modal doesn't appear.
> > > Screen.Recording.2022-10-12.at.10.49.34.AM.mov
> > 
> > 
> > @jepstein The "check your email" modal is supposed to appear after you hit "Resend" on the "verify your email" toast. Could you please try pressing the "Resend" button on the toast?
> 
> Aha! Didn't realize that's how it worked. Sorry for the misdirection. It's working just as expected.

YAY! Thanks for testing

**Review by thomik-corp — APPROVED** [review]
Looks good. We can remove the icon-checkmark.svg and I made a simple spelling suggestion. Rest looks good to me.

**Comment by Carla-Moz** [comment]
> Looks good. We can remove the icon-checkmark.svg and I made a simple spelling suggestion. Rest looks good to me.

Thank you! I'll add the tests when I refactor the Product Toast file.

**Pull request merged**